code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/HostValidationFilter.java

```diff
@@ -0,0 +1,56 @@
+package io.cattle.platform.servicediscovery.api.filter;
+
+import io.cattle.platform.api.utils.ApiUtils;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.iaas.api.infrastructure.InfrastructureAccessManager;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManagerLocator;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class HostValidationFilter extends AbstractDefaultResourceManagerFilter {
+
+    @Inject
+    ResourceManagerLocator locator;
+    @Inject
+    ObjectManager objMgr;
+    @Inject
+    InfrastructureAccessManager infraAccess;
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { Host.class };
+    }
```

Comment:
The actions actually automatically get validation based off of their presence/absence as controlled by capability attribute. Its baked into the framework.
So for example, if a restricted user attempts to hit `/hosts/1h1?action=deactivate` they get a 404 because they cannot see that action.